R/openapi.R

```diff
@@ -105,3 +105,97 @@ json_type <- function(type) {
 }
 
 
+# Generate an informational message based on the execution context of the extension
+warning_message <- function() {
+  message_contents <- NULL
+  # RStudio Connect Details
+  # Provide messaging if RSC:
+  #  * Is a version that doesn't support Tableau Extensions
+  #  * Isn't configured to support Tableau Extensions
+  #  * Doesn't have Server.Address configured
+  # TODO: Replace this with proper logic once available
+
+  # Server.Address
+  connect_server <- Sys.getenv("CONNECT_SERVER")
+
+  # Does this installation support Tableau Extensions
+  connect_support <- Sys.getenv("RSC_TABLEAU")
+
+  # RStudio Connect version
+  connect_version <- Sys.getenv("RSC_VERSION")
+
+  if (connect_version != "1.9.0") {
+    message_contents <- paste(message_contents,
+                              "* **This version of RStudio Connect does not support Tableau Analytics Extension APIs. Please upgrade RStudio Connect to at least version 1.9.x**",
+                              sep = "\n")
+  }
+  if (!rlang::is_true(as.logical(connect_support))) {
+    message_contents <- paste(message_contents,
+                              "* This installation of RStudio Connect does not currently support Tableau Analytics Extension APIs.",
+                              sep = "\n")
+  }
+
+  if (connect_server == "") {
+    message_contents <- paste(message_contents,
+                              "* The `Server.Address` property isn't configured for this installation of RStudio Connect.",
+                              sep = "\n")
+  }
+
+  # Send warning message to the console if any of the above are TRUE
+  if (!rlang::is_null(message_contents)) {
+    message_contents <- paste(
+      message_contents,
+      "\n\n#### Please reach out to your RStudio Connect administrator",
+      sep = "\n"
+    )
+
+    rlang::warn(stringi::stri_replace_all(message_contents, regex = "\\* ?", replacement = ""), .frequency = "once", .frequency_id = "rsc_warning")
+  }
+
+  message_contents
+}
+
+
+info_message <- function() {
+  message_contents <- NULL
+  if (stringi::stri_detect(Sys.getenv("DEBUGME"), fixed = "plumbertableau")) {
+    message_contents <- paste(message_contents,
+                              "* Debugging is **on**.
+  * To disable it please remove the `DEBUGME` environment variable or set it to a value that does not include 'plumbertableau'.",
+  sep = "\n")
+  } else {
+    message_contents <- paste(message_contents,
+                              "* Debugging is **off**.
+  * To enable it please set the environment variable `DEBUGME` to include 'plumbertableau'.",
+  sep = "\n")
+  }
+
+  message_contents
+}
+
+
+# Generate message for extension setup
+extension_setup <- function() {
```

Comment:
Since the setup instructions are moving to their own page, it may make sense to move this outside of `openapi.R`